src/client/layout/layout_object.cpp

```diff
@@ -196,7 +196,16 @@ namespace client_layout
       if (webContent != nullptr)
       {
         webContent->setLayer(layer_);
-        webContent->setIsScrollableContainer(isScrollContainer());
+        if (isScrollContainer())
+        {
+          webContent->setIsScrollableContainer(true);
+          webContent->setBelongsToScrollableContainer(entity_.value());
+        }
+        else
+        {
+          webContent->setIsScrollableContainer(false);
+          webContent->setBelongsToScrollableContainer(containingScrollContainer()->entity());
```

Comment:
Could you please optimize `containingScrollContainer()`, now it walks from the current node up until found a scrollable container, may you introduce a cache for this?